datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -1654,7 +1636,8 @@ async fn output_single_parquet_file_parallelized(
     object_store_writer: Box<dyn AsyncWrite + Send + Unpin>,
     data: Receiver<RecordBatch>,
     output_schema: Arc<Schema>,
-    parquet_props: &WriterProperties,
+    writer_properties: &WriterProperties,
+    skip_arrow_metadata: bool,
```

Comment:
I  don't follow why this new `skip_arrow_metadata` parameter is needed. It looks like this wasn't used previously so is this a change in behaviour where previously this option was ignored?

Reply:
I'm still confused. That other path was already used without encryption, so why would this only be needed for encrypted files?

Reply:
Removed some schema logic https://github.com/apache/datafusion/pull/16738/commits/915bfee52a095e11f454a2451c3b6b9cc8cadf02